### ISSUE TITLE
the components do not re-render when the props 'value' changed . 

### ISSUE THREAD

**Issue by westpsk** [Vnkitaev/react-summernote#17]
Here is how I use : 

The console log of `defaultValue` is right, but when it changes , the component does not re-render.
the textarea stay the last value.
 
```

render() {
    const { defaultValue } = this.props
    console.log("defaultValue")
    console.log(defaultValue)
    return (
      <ReactSummernote
        value={defaultValue}
        options={{
          lang: 'zh-CN',
          height: 150,
          dialogsInBody: true,
          toolbar: [
            ['style', ['style']],
            ['font', ['bold', 'underline', 'clear']],
            ['fontname', ['fontname']],
            ['para', ['ul', 'ol', 'paragraph']],
            ['table', ['table']],
            ['insert', ['link', 'picture']],
            ['view', ['fullscreen', 'codeview']]
          ]
        }}
        onChange={this.onChange}
      />
    )
  }
```

**Comment by alameya** [comment]
it happens bacause one clever man did that 
```javascript
shouldComponentUpdate() {
    return false;
  }
```
remove that and be happy

**Comment by Vnkitaev** [comment]
Hi @westpsk, the `value` is controlled inside the jQuery Summernote plugin, thus if the component will be re-rendered, the plugin may unbind.
 
So in order to avoid this, as @alameya mentioned, the `shouldComponentUpdate` method returns `false`, so the component will not be re-rendered and the plugin would not unbind. Otherwise the jQuery plugin should be initialized again with the new `value` in `componentDidUpdate` on every re-render, which is not that efficient.
 
__TL DR: The solution to your problem is this:__
 
There is an `onChange` prop, through this callback you receive every new Summernote's `value`, which you store then. So the component should not be re-rendered and the `value` you have is always up-to-date.
 
If you want to have all the control over the value and the `defaultValue` prop in this component, you are always welcome to send PR.

**Comment by alameya** [comment]
@westpsk, ok. look to the next case:
I have SPA with your plugin for creating a new records.
After submit of course if will get value but i need re-render component with empty value.
if plugin doesn't work properly it is poor than if will it work slowly

**Comment by alxrm** [comment]
@alameya, thank you for such an attention to this library, but If you'll read the the docs again and just a bit more carefully, you would notice the title:

 # Static methods

and then the first method, you'll probably see there, might be

```js
reset() // Clear contents and remove all stored history
```

which solves the case you've described above. Hope this helps.

Best regards!

**Comment by Vnkitaev** [comment]
Thank you, @alxrm for clarification
 
@alameya, as I mentioned earlier, if you want some more functionality like `defaultValue` or full control over the `value`, you are always free to send PR's here.
 
Otherwise, you have to be patient and wait for me to implement those features.

**Comment by westpsk** [comment]
@alameya @alameya  Thanks for your patience.  @alameya I tried the two method you said, remove  `shouldComponentUpdate() { return false; }` and re-build it . or add `shouldComponentUpdate={() => {}}` in the component.

 But It seems does not work on me, the component does not re-render ... 


```

import React, { Component } from 'react'
import ReactSummernote from 'react-summernote'
import 'react-summernote/dist/react-summernote.css' // import styles
import 'react-summernote/lang/summernote-zh-CN' // you can import any other locale

// Import bootstrap(v3 or v4) dependencies
import 'bootstrap/js/modal'
import 'bootstrap/js/dropdown'
import 'bootstrap/js/tooltip'
import '../../style/bootstrap.css'
class RichTextEditor extends Component {
  
  constructor(props, context) {
    super(props, context)
    this.state = {
      richText: null
    }
  }

  onChange = (content) => {
    this.setState({
      richText: content
    })
    this.props.callbackParent(content)
  }

  render() {
    const { defaultValue } = this.props
    console.log("defaultValue")
    console.log(defaultValue)
    return (
      <ReactSummernote
        value={defaultValue}
        options={{
          lang: 'zh-CN',
          height: 150,
          dialogsInBody: true,
          toolbar: [
            ['style', ['style']],
            ['font', ['bold', 'underline', 'clear']],
            ['fontname', ['fontname']],
            ['para', ['ul', 'ol', 'paragraph']],
            ['table', ['table']],
            ['insert', ['link', 'picture']],
            ['view', ['fullscreen', 'codeview']]
          ]
        }}
        onChange={this.onChange}
        shouldComponentUpdate={() => {}}
      />
    );
  }
}

export default RichTextEditor

```

**Comment by lorencole** [comment]
@alxrm @Vnkitaev 

Can you provide and example of how to use reset() from the parent component? I can get my hands on ReactSummernote using a ref as show below,  but reset() is not available.

`<ReactSummernote options={_editorProps} onChange={ this.handleChange.bind(this) } value={ input.value } ref={(summernote) => {this.summernote = summernote}} />`

**Comment by Vnkitaev** [comment]
react-summernote 2.0.0 support controlled value and uncontrolled defaultValue.

**Comment by meetalodariya** [comment]
A simple solution is to add onInit callback in component 

`<ReactSummernote
        options={{
          height: 350,
          dialogsInBody: true,
          toolbar: [
            ["style", ["style"]],
            ["font", ["bold", "italic", "underline", "clear"]],
            ["fontname", ["fontname", "fontsize", "fontsizeunit", "color"]],
            ["para", ["ul", "ol", "paragraph", "height"]],
            ["table", ["table"]],
            ["insert", ["link"]],
            ["view", ["codeview"]],
            ["hr"],
          ],
        }}
        **onInit={() => {
          const editArea = document.querySelector(".note-editable");
          editArea.innerHTML = templateHtml;
        }}**
        onChange={(value) => {
          draftCreateTemplateFieldChange("templateHtml", value);
        }}
      />`

**Comment by meetalodariya** [comment]
A simple solution is to add onInit callback in component and set the html value of the text node.



`
<ReactSummernote
        options={{
          height: 350,
          dialogsInBody: true,
          toolbar: [
            ["style", ["style"]],
            ["font", ["bold", "italic", "underline", "clear"]],
            ["fontname", ["fontname", "fontsize", "fontsizeunit", "color"]],
            ["para", ["ul", "ol", "paragraph", "height"]],
            ["table", ["table"]],
            ["insert", ["link"]],
            ["view", ["codeview"]],
            ["hr"],
          ],
        }}
        onInit={() => {
          const editArea = document.querySelector(".note-editable");
          editArea.innerHTML = templateHtml;
        }}
        onChange={(value) => {
          draftCreateTemplateFieldChange("templateHtml", value);
        }}
      />

`

**Comment by meetalodariya** [comment]
A simple solution is to add onInit callback in component and set the html value of the text node.

`    
    <ReactSummernote    
      onInit={() => {
          const editArea = document.querySelector(".note-editable");
          editArea.innerHTML = {your value};
        }}
     .....
      >
`
